docs/security.md

```diff
@@ -563,7 +604,52 @@ replaced with one of the above namespaces.
   <tr>
     <td><code>${ns}.trustStoreType</code></td>
     <td>JKS</td>
-    <td>The type of the trust store.</td>
+    <td>The type of the trust store. This setting is not applicable to the `rpc` namespace.</td>
+  </tr>
+  <tr>
+    <td><code>${ns}.openSSLEnabled</code></td>
+    <td>false</td>
+    <td>
+      Whether to use OpenSSL for cryptographic operations instead of the JDK SSL provider.
+      This setting is only applicable to the `rpc` namespace, and also requires the `certChain`
+      and `privateKey` settings to be set.
```

Comment:
Call out that this takes precedence over JKS, and that unavailability of openssl at runtime will cause spark to fall back to jks ?

Reply:
Documented. Updating the PR in a second..

> QQ: Would be interesting to see what would happen in case jks and openssl configs are not 'compatible' (if jks was being specified for UI) - how spark behaves: perhaps we should disable that fallback ?

Could you clarify what you mean? 

I see a few cases:

1. OpenSSL requested, and privateKey/certChain not set in the configs. This is an error and will throw saying keys are missing.
2. OpenSSL not requested, and JKS keys are specified (keyStore/trustStore). The JKS keys will be used
3. OpenSSL requested, but not available at runtime. This uses the JDK SSL implementation, but with the keys specified in `certChain` and `privateKey`.

Was that what you were asking about?

Reply:
I am referring to (3).
If openssl is requested but is not available at runtime, we will try to fallback to jks.
The openssl config for `privateKey`/`certChain` need not be compatible with the `certChain`/`privateKey` for jks - for example, openssl for rpc vs jks for ui.

Reply:
@mridulm I might be missing something. We clearly document two things that I believe should mean we don't have a problem:

1. We say `privateKey` and `certChain` must be PEM files,
2. We say that `privateKey` and `certChain` are only supported for the rpc namespace.

If one wants to use `keyStore` and `trustStore` it *must* be a JKS file, and you then must use the JDK SSL provider (not openssl). In that case the keys will be the same across rpc and UI and things should work fine.

Reply:
You are right, it is namespaced to rpc, and so won't use the keystore from other namespace ... sigh, should have reread that better.

Reply:
no worries! this is confusing and I am open to suggestions on how to make this clearer since if it's confusing to you it might be confusing for others. We can iterate on the docs if people have confusion going forward.